markdown/events/2021/bytesize-7-nf-core-ci-tests.md

```diff
@@ -34,3 +34,256 @@ The talk will be presented on Zoom and live-streamed on YouTube:
+
+[19:13](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1153) So I can do a `git pull upstream master`, and it tells me that I have a merge conflict.
+
+[19:20](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1160) So if I hop into vs code, which is here, you can see the `merge conflicts` and `changelog`.
+
+Sure enough, there are two lines, which have been added kind of in parallel, and git does not know how to merge in both. But, it’s just a `changelog`. We want both of these [19:33](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1173).
+
+[19:34](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1174) So I can click accept both.
+
+[19:37](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1177) There are some differences in the markdown styles here, so hopefully if I hit save, it will solve that.
+
+[19:43](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1183) I can go back and `fix merge conflict`. This will do several things. It’s fixing a merge conflict so the pull request test will run. It’s also pushing a new change after I changed the target branch to dev.
+
+[20:00](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1200) So the target branch test should now be passing, and we’ll see how much else is green.
+
+[20:08](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1208) This is the point where I suspect we won’t get green across the board because of running on the development version of nf-core. I think we didn’t manage to scrape up every single test, but hopefully by the time you run this you will get all green tests at this point.
+
+[20:39](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1239) So what I haven’t talked about so far is running workflow tests.
+
+[20:43](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1243) So every nf-core pipeline has a way you can run it where you do `nextflow run pipeline name` and then you do `-profile test`. That runs the pipeline with a very tiny test dataset, which is downloaded from the web dynamically by an expert.
+
+[20:57](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1257) So now basically, every single time you push or open a pull request, this test profile will be run.
+
+[21:05](https://youtu.be/U9LG_mMQFMY?list=PL3xpfTVZLcNiSvvPWORbO32S1WDJqKp1e&t=1265) That just run the NextFlow pipeline and checks that it doesn’t crash and picks up a bunch of kind of potential problems here.
+
+I’ve forgotten here that these tests are also quite intelligent. They check whether you have changed anything in the environment (the Docker file or Conda). If you have, then it will build a new docker image before running the pipeline. If you haven’t, it will just pull it from docker, which is a lot quicker.
```

Comment:
```suggestion
I’ve forgotten here that these tests are also quite intelligent. They check whether you have changed anything in the environment (the Docker file or Conda). If you have, then it will build a new Docker image before running the pipeline. If you haven’t, it will just pull it from Docker, which is a lot quicker.
```